### ISSUE TITLE
Bicolor Status Bar Bug

### ISSUE THREAD

**Issue by danilobuerger** [wix/react-native-navigation#6506]
### Issue Description

Multiple calls to `Navigation.setRoot` lead to the status bar becoming bicolor (text white, icons black).

Before             |  After
:-------------------------:|:-------------------------:
![Simulator Screen Shot - iPhone SE (2nd generation) - 2020-08-24 at 15 59 16](https://user-images.githubusercontent.com/996231/91053214-cbc23380-e622-11ea-8612-1d28649c2a94.png)  |  ![Simulator Screen Shot - iPhone SE (2nd generation) - 2020-08-24 at 15 59 08](https://user-images.githubusercontent.com/996231/91053235-d1b81480-e622-11ea-9049-3321083688b3.png)

### Steps to Reproduce / Code Snippets / Screenshots

Easily reproducible with current playground (https://github.com/wix/react-native-navigation/tree/6022f434a8b4cd4a1b03f1f9f2a13e6912b7e07e):

```diff
diff --git a/playground/src/app.ts b/playground/src/app.ts
index dbe0ce1a7..4907ee482 100644
--- a/playground/src/app.ts
+++ b/playground/src/app.ts
@@ -26,10 +26,17 @@ function start() {
   setDefaultOptions();
   Navigation.events().registerAppLaunchedListener(async () => {
     Navigation.dismissAllModals();
-    setRoot();
+    repeatedlyCallSetRoot();
   });
 }
 
+function repeatedlyCallSetRoot() {
+  setRoot();
+  setTimeout(() => {
+    repeatedlyCallSetRoot();
+  }, 5000);
+}
+
 function setRoot() {
   Navigation.setRoot({
     root: {

```

Now `Navigation.setRoot` is called every 5 seconds in the playground app, leading to the bicolor status bar.

---
### Environment
* React Native Navigation version: https://github.com/wix/react-native-navigation/tree/6022f434a8b4cd4a1b03f1f9f2a13e6912b7e07e
* React Native version: 0.61.4, 0.62.2, 0.63.2
* Platform(s) (iOS, Android, or both?): iOS
* Device info (Simulator/Device? OS version? Debug/Release?): iOS Simulator (iOS 13.6, XCode 11.6). Also iPhone SE 2nd (iOS 13.6, XCode 11.6, release)

**Comment by danilobuerger** [comment]
Modifying `RNNBasePresenter` results in it not happening any more:

```
diff --git a/lib/ios/RNNBasePresenter.m b/lib/ios/RNNBasePresenter.m
index a4bf12294..f3fc3a099 100644
--- a/lib/ios/RNNBasePresenter.m
+++ b/lib/ios/RNNBasePresenter.m
@@ -90,19 +90,7 @@ - (void)viewDidLayoutSubviews {
 }
 
 - (UIStatusBarStyle)getStatusBarStyle {
-    RNNNavigationOptions *withDefault = [self.boundViewController.resolveOptions withDefault:[self defaultOptions]];
-    NSString* statusBarStyle = [withDefault.statusBar.style getWithDefaultValue:@"default"];
-    if ([statusBarStyle isEqualToString:@"light"]) {
-        return UIStatusBarStyleLightContent;
-    } else if (@available(iOS 13.0, *)) {
-        if ([statusBarStyle isEqualToString:@"dark"]) {
-            return UIStatusBarStyleDarkContent;
-        } else {
-            return UIStatusBarStyleDefault;
-        }
-    } else {
-        return UIStatusBarStyleDefault;
-    }
+    return UIStatusBarStyleDarkContent;
 }
 
 - (UINavigationItem *)currentNavigationItem {
@@ -114,12 +102,6 @@ - (UIInterfaceOrientationMask)getOrientation {
 }
 
 - (BOOL)getStatusBarVisibility {
-    RNNNavigationOptions *withDefault = [self.boundViewController.resolveOptions withDefault:self.defaultOptions];
-    if (withDefault.statusBar.visible.hasValue) {
-        return ![withDefault.statusBar.visible get];
-    } else if ([withDefault.statusBar.hideWithTopBar getWithDefaultValue:NO]) {
-        return self.boundViewController.stack.isNavigationBarHidden;
-    }
     return NO;
 }
 
```

**Comment by danilobuerger** [comment]
It seems to be timing related as doing a NSThread sleep inside `getStatusBarStyle` makes it reproducible again:
```diff
- (UIStatusBarStyle)getStatusBarStyle {
    [NSThread sleepForTimeInterval:0.01f];
    return UIStatusBarStyleDarkContent;
}
```
